clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -228,7 +229,8 @@ private void poll(long timeout, long now, boolean executeDelayedTasks) {
         trySend(now);
 
         // fail all requests that couldn't be sent
-        failUnsentRequests();
+        if (failUnsent)
+            failUnsentRequests();
```

Comment:
@hachikuji Thank you for the review. I was not sure of the implications of not failing unsent requests. Hence tried to keep the code as close to the existing semantics as possible. I like the idea of only expiring requests on timeout. Are coordinator changes always associated with a connection failure? If they are, then it shouldn't be hard, otherwise it may be harder to ensure all the edge cases are handled.